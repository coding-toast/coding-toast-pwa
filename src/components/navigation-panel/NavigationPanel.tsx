import * as React from 'react';
import { Navbar, Nav } from 'react-bootstrap';
import Link from 'next/link';
import { Logo } from 'components';
import RouteLink from './RouteLink';

const pageLinks = [
  {
    href: '/blog',
    label: 'Blog'
  },
  {
    href: '/resources',
    label: 'Resources'
  },
  {
    href: '/contact',
    label: 'Contact'
  }
];

const navLinks = pageLinks.map((element) => <RouteLink key={element.href} href={element.href} label={element.label} />);

const NavigationPanel: React.FC = () => (
  <Navbar collapseOnSelect expand={false} fixed='top' sticky='top' bg='dark' variant='dark' className='bg-background-primary'>
    <Navbar.Brand>
      <Link href='/'>
        <Logo />
      </Link>
    </Navbar.Brand>
    <Navbar.Toggle aria-controls='responsive-navbar-nav' />
    <Navbar.Collapse id='responsive-navbar-nav'>
      <Nav className='text-right'>{navLinks}</Nav>
    </Navbar.Collapse>
  </Navbar>
);

export default NavigationPanel;
