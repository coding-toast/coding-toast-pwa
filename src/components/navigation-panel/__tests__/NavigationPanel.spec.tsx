import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavigationPanel from '../NavigationPanel';

describe('<NavigationPanel />', () => {
  it('renders correctly', () => {
    const component = shallow(<NavigationPanel />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
