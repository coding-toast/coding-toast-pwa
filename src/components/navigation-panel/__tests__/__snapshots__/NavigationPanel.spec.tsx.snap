// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly 1`] = `
<Navbar
  bg="dark"
  className="bg-background-primary"
  collapseOnSelect={true}
  expand={false}
  fixed="top"
  sticky="top"
  variant="dark"
>
  <NavbarBrand>
    <mockConstructor
      to="/"
    >
      <Logo />
    </mockConstructor>
  </NavbarBrand>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Nav
      className="text-right"
      fill={false}
      justify={false}
    >
      <NavItem>
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
        >
          <mockConstructor
            className="text-text-color-primary"
            to="/m"
          >
            Blog
          </mockConstructor>
        </NavLink>
      </NavItem>
      <NavItem>
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
        >
          <mockConstructor
            to="/resources"
          >
            Resources
          </mockConstructor>
        </NavLink>
      </NavItem>
      <NavItem>
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
        >
          <mockConstructor
            className="text-text-color-primary"
            to="/contact"
          >
            Contact
          </mockConstructor>
        </NavLink>
      </NavItem>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;
